### PR TITLE
 Allow requesting more than 65536 bytes in the browser, with no copying

### PR DESCRIPTION
**Pull request by calvinmetcalf**
based on #15

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -15,19 +23,23 @@ if (crypto && crypto.getRandomValues) {
 
 function randomBytes (size, cb) {
   // phantomjs needs to throw
-  if (size > 65536) throw new Error('requested too many random bytes')
+  if (size > MAX_UINT32) throw new Error('requested too many random bytes')
```
Comment (dcousens) [inline browser.js]: `RangeError`?

File: browser.js [hunk 2]
```diff
@@ -15,19 +23,23 @@ if (crypto && crypto.getRandomValues) {
...
   // in case browserify  isn't using the Uint8Array version
-  var rawBytes = new global.Uint8Array(size)
+  var bytes = Buffer.allocUnsafe(size)
 
   // This will not work in older browsers.
   // See https://developer.mozilla.org/en-US/docs/Web/API/window.crypto.getRandomValues
   if (size > 0) {  // getRandomValues fails on IE if size == 0
-    crypto.getRandomValues(rawBytes)
+    if (size > MAX_BYTES) {
```
Comment (dcousens) [inline browser.js]: Is it `> MAX_BYTES`, or `>= MAX_BYTES`?

**Comment by achingbrain** [comment]
Do you think this will be released any time soon?

**Comment by alanshaw** [comment]
+1

**Comment by achingbrain** [comment]
@calvinmetcalf is there anything you're waiting for before publishing this?  It'd be great if this module had the same behaviour as node.

**Comment by calvinmetcalf** [comment]
ok once these test pass I'll merge and publish

**Pull request merged**